rskj-core/src/test/java/co/rsk/scoring/InetAddressCidrBlockTest.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2022 RSK Labs Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.scoring;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+/**
+ * Created by ajlopez on 11/07/2017.
+ */
+public class InetAddressCidrBlockTest {
+
+    @Test
+    public void equals() throws UnknownHostException {
+        InetAddress address1 = InetAddress.getByName("192.168.0.47");
+        InetAddress address2 = InetAddress.getByName("198.168.0.47");
+        InetAddress address3 = InetAddress.getByName("2001:0db8:85a3:0000:0000:8a2e:0370:7335");
+
+        InetAddressCidrBlock block1 = new InetAddressCidrBlock(address1, 8);
+        InetAddressCidrBlock block2 = new InetAddressCidrBlock(address2, 9);
+        InetAddressCidrBlock block3 = new InetAddressCidrBlock(address1, 1);
+        InetAddressCidrBlock block4 = new InetAddressCidrBlock(address1, 8);
+        InetAddressCidrBlock block5 = new InetAddressCidrBlock(address3, 8);
+
+        Assert.assertEquals(block1, block1);
+        Assert.assertEquals(block2, block2);
+        Assert.assertEquals(block3, block3);
+        Assert.assertEquals(block4, block4);
+        Assert.assertEquals(block5, block5);
+
+        Assert.assertEquals(block1, block4);
+        Assert.assertEquals(block4, block1);
+
+        Assert.assertNotEquals(block1, block2);
+        Assert.assertNotEquals(block1, block3);
+        Assert.assertNotEquals(block1, block5);
+
+        Assert.assertNotEquals(null, block1);
+
+        Assert.assertEquals(block1.hashCode(), block4.hashCode());
+    }
+
+    @Test
+    public void subnetWithCidr8IPV4() throws UnknownHostException {

```

Comment:
could you also add a test for cidr `/0` ?

Reply:
Missed that one, thanks. Fixed.